osu.Game/Updater/AssociationManager.cs

```diff
@@ -0,0 +1,168 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using System;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+using System.Runtime.InteropServices;
+using System.Runtime.Versioning;
+using Microsoft.Win32;
+using osu.Framework.Allocation;
+using osu.Framework.Graphics.Containers;
+using osu.Framework.Graphics.Sprites;
+using osu.Framework.Logging;
+using osu.Game.Overlays;
+using osu.Game.Overlays.Notifications;
+
+namespace osu.Game.Updater
+{
+    [SupportedOSPlatform("windows")]
+    public partial class AssociationManager : CompositeDrawable
+    {
+        // Needed so that Explorer windows get refreshed after the registry is updated
+        [DllImport("Shell32.dll")]
+        private static extern int SHChangeNotify(int eventId, int flags, IntPtr item1, IntPtr item2);
+
+        private static readonly string[] associated_extensions =
+        {
+            ".osz",
+            ".osr",
+            ".osk",
+            ".osu",
+            ".osr",
+        };
```

Comment:
Probably shouldn't be hard-coded. And if it is, it should match what is actually supported. Namely: `.olz` is missing, and `.osu` doesn't seem to be supported by direct imports.

https://github.com/search?q=repo%3Appy%2Fosu+%22HandledExtensions+%3D%3E+new%5B%5D%22&type=code

Reply:
Added olz.

As for .osu, I think it might be a better idea to add a way to create a naming for it but not associate it with the application (i.e. don't add the shell\open\command, only set the default value). From what I've observed, this is what stable does.